lib/Dialect/FIRRTL/LowerTypes.cpp

```diff
@@ -276,8 +295,13 @@ void FIRRTLTypesLowering::visitStmt(ConnectOp op) {
   Value dest = op.dest();
   Value src = op.src();
 
+  // Attempt to get the bundle types, potentially unwrapping an outer flip type
+  // that wraps the whole bundle.
+  BundleType destType = getCanonicalBundleType(dest.getType());
+  BundleType srcType = getCanonicalBundleType(src.getType());
+
   // If we aren't connecting two bundles, there is nothing to do.
-  if (!dest.getType().isa<BundleType>() || !src.getType().isa<BundleType>())
+  if (!destType || !srcType)
```

Comment:
@lattner I think I finally understand your question here about outer level flip types: https://github.com/llvm/circt/pull/157#discussion_r519233669. I was getting bit here trying to write this [test case](https://github.com/llvm/circt/pull/263/files#diff-cfcb11e78479b63cf6e909ee4a9e01a0ff1ba2ab019da32a0c64baa56b7e0c12R143). I've pulled the logic to peel off the outer flip so I could use it here as well.